reactor-aeron-core/src/main/java/reactor/aeron/MessagePublication.java

```diff
@@ -43,130 +49,123 @@
     this.connectTimeout = options.connectTimeout();
     this.backpressureTimeout = options.backpressureTimeout();
     this.adminActionTimeout = options.adminActionTimeout();
-    this.writeLimit = options.resources().writeLimit();
   }
 
   /**
-   * Enqueues content for future sending.
+   * Enqueues publisher for future sending.
    *
-   * @param buffer abstract buffer to send
+   * @param publisher abstract publisher to process messages from
    * @param bufferHandler abstract buffer handler
    * @return mono handle
    */
-  <B> Mono<Void> publish(B buffer, DirectBufferHandler<? super B> bufferHandler) {
+  <B> Mono<Void> publish(Publisher<B> publisher, DirectBufferHandler<? super B> bufferHandler) {
     return Mono.defer(
         () -> {
-          if (isDisposed()) {
-            return Mono.error(AeronExceptions.failWithPublicationUnavailable());
-          }
-          MonoProcessor<Void> promise = MonoProcessor.create();
-          if (!publishTasks.offer(
-              PublishTask.newInstance(publication, buffer, bufferHandler, promise))) {
-            return Mono.error(AeronExceptions.failWithPublicationUnavailable());
-          }
-          return promise;
+          PublisherProcessor processor = new PublisherProcessor(bufferHandler, this);
+          publisher.subscribe(processor);
+          return processor.onDispose();
         });
   }
 
   /**
-   * Proceed with processing of tasks.
+   * Makes a progress at processing publisher processors collection. See for details {@link
+   * PublisherProcessor}.
    *
    * @return more than or equal {@code 1} - some progress was done; {@code 0} - denotes no progress
    *     was done
    */
-  int proceed() {
+  int publish() {
+
+    PublisherProcessor[] oldArray = this.publisherProcessors;
```

Comment:
it's worse name let's rename it to `currentProcessors`

Reply:
I like `oldArray`